src/mood.ts

```diff
@@ -303,6 +310,70 @@ class AsdonMoodElement extends MoodElement {
   }
 }
 
+class ShieldMoodElement extends MoodElement {
+  effect: Effect;
+  skill: Skill;
+
+  constructor(effect: Effect) {
+    super();
+    this.effect = effect;
+    const skill = shieldEffects.get(effect);
+    if (!skill) throw `Invalid shield effect: ${effect}`;
+    this.skill = skill;
+  }
+
+  mpCostPerTurn(): number {
+    const turns = turnsPerCast(this.skill);
+    return turns > 0 ? mpCost(this.skill) / turns : 0;
+  }
+
+  turnIncrement(): number {
+    return turnsPerCast(this.skill);
```

Comment:
I don't think we can rely on turnsPerCast here, we might need to add it to the map